ApacheThriftPyGenTest.test_namespace_effective fails when run in isolation

Looks like some kind of `sys.path` isolation issue. It passes when run with other tests.

```
$ ./pants test tests/python/pants_test/backend/codegen/thrift/python -- -s -k test_namespace_effective
   Compiling engine v0.0.1 (file:///Users/dwagnerhall/src/github.com/pantsbuild/pants/src/rust/engine)
    Finished dev [unoptimized + debuginfo] target(s) in 34.6 secs
WARN] Globs did not match. Excludes were: []. Unmatched globs were: ["tests/python/pants_test/backend/codegen/thrift/python/*_test.py"].

13:27:52 00:00 [main]
               (To run a reporting server: ./pants server)
13:27:53 00:01   [setup]
13:28:10 00:18     [parse]
               Executing tasks in goals: bootstrap -> imports -> unpack-jars -> jvm-platform-validate -> deferred-sources -> gen -> resolve -> resources -> pyprep -> compile -> test
13:28:11 00:19   [bootstrap]
13:28:11 00:19     [substitute-aliased-targets]
13:28:11 00:19     [jar-dependency-management]
13:28:11 00:19     [bootstrap-jvm-tools]
13:28:11 00:19     [provide-tools-jar]
13:28:11 00:19   [imports]
13:28:11 00:19     [ivy-imports]
13:28:11 00:19   [unpack-jars]
13:28:11 00:19     [unpack-jars]
13:28:11 00:19   [jvm-platform-validate]
13:28:11 00:19     [jvm-platform-validate]
13:28:11 00:19   [deferred-sources]
13:28:11 00:19     [deferred-sources]
13:28:11 00:19   [gen]
13:28:11 00:19     [antlr-java]
13:28:11 00:19     [antlr-py]
13:28:11 00:19     [jaxb]
13:28:11 00:19     [protoc]
13:28:11 00:19     [ragel]
13:28:11 00:19     [thrift-java]
13:28:11 00:19     [thrift-py]
13:28:11 00:19     [wire]
13:28:11 00:19     [avro-java]
13:28:12 00:20     [go-thrift]
13:28:12 00:20     [go-protobuf]
13:28:12 00:20     [jax-ws]
13:28:12 00:20     [scrooge]
13:28:12 00:20     [thrifty]
13:28:12 00:20   [resolve]
13:28:12 00:20     [ivy]
13:28:12 00:20     [coursier]
13:28:12 00:20     [go]
13:28:12 00:20     [scala-js-compile]
13:28:12 00:20     [scala-js-link]
13:28:12 00:20     [node]
13:28:12 00:20   [resources]
13:28:12 00:20     [prepare]
13:28:12 00:20     [services]
13:28:12 00:20   [pyprep]
13:28:12 00:20     [interpreter]
13:28:12 00:20     [build-local-dists]
13:28:12 00:20     [requirements]
13:28:27 00:35     [sources]
                   Invalidated 6 targets.
13:28:29 00:37   [compile]
13:28:29 00:37     [node]
13:28:29 00:37     [compile-jvm-prep-command]
13:28:29 00:37       [jvm_prep_command]
13:28:29 00:37     [compile-prep-command]
13:28:29 00:37     [compile]
13:28:29 00:37     [zinc]
13:28:29 00:37     [javac]
13:28:29 00:37     [cpp]
13:28:29 00:37     [errorprone]
13:28:29 00:37     [findbugs]
13:28:30 00:38     [go]
13:28:30 00:38   [test]
13:28:30 00:38     [test-jvm-prep-command]
13:28:30 00:38       [jvm_prep_command]
13:28:30 00:38     [test-prep-command]
13:28:30 00:38     [test]
13:28:30 00:38     [pytest-prep]
                   Invalidated 46 targets.
13:28:30 00:38     [pytest]
                   Invalidated 1 target.
                   scrubbed PYTHONPATH=/Users/dwagnerhall/src/github.com/pantsbuild/pants/src/python: from py.test environment
13:28:30 00:38       [run]
                     ============== test session starts ===============
                     platform darwin -- Python 2.7.14, pytest-3.4.2, py-1.5.2, pluggy-0.6.0
                     rootdir: /Users/dwagnerhall/src/github.com/pantsbuild/pants/.pants.d, inifile: /Users/dwagnerhall/src/github.com/pantsbuild/pants/.pants.d/test/pytest-prep/CPython-2.7.14/e4b37e66c6edb9ee569b86090a0be4a103072a92/pytest.ini
                     plugins: timeout-1.2.1, cov-2.4.0
                     
                     Starting workunit execute
                     
                     Starting workunit src/thrift/com/foo/one.thrift
                     
                     Starting workunit execute
                     
                     Starting workunit src/thrift2/com/foo/two.thrift
                     collected 3 items
                     
                     tests/python/pants_test/backend/codegen/thrift/python/test_apache_thrift_py_gen.py F
                     
                      generated xml file: /Users/dwagnerhall/src/github.com/pantsbuild/pants/.pants.d/test/pytest/tests.python.pants_test.backend.codegen.thrift.python.python/junitxml/TEST-tests.python.pants_test.backend.codegen.thrift.python.python.xml 
                     ==================== FAILURES ====================
                     _ ApacheThriftPyGenTest.test_namespace_effective _
                     
                     self = <pants_test.backend.codegen.thrift.python.test_apache_thrift_py_gen.ApacheThriftPyGenTest testMethod=test_namespace_effective>
                     
                         def test_namespace_effective(self):
                           self.create_file('src/thrift/com/foo/one.thrift', contents=dedent("""
                             namespace py foo.bar
                         
                             struct One {}
                             """))
                           one = self.make_target(spec='src/thrift/com/foo:one',
                                                  target_type=PythonThriftLibrary,
                                                  sources=['one.thrift'])
                           apache_thrift_gen, synthetic_target_one = self.generate_single_thrift_target(one)
                         
                           self.create_file('src/thrift2/com/foo/two.thrift', contents=dedent("""
                             namespace py foo.baz
                         
                             struct Two {}
                             """))
                           two = self.make_target(spec='src/thrift2/com/foo:two',
                                                  target_type=PythonThriftLibrary,
                                                  sources=['two.thrift'])
                           _, synthetic_target_two = self.generate_single_thrift_target(two)
                         
                           # Confirm separate PYTHONPATH entries, which we need to test namespace packages.
                           self.assertNotEqual(synthetic_target_one.target_base, synthetic_target_two.target_base)
                         
                           targets = (synthetic_target_one, synthetic_target_two)
                         
                           python_repos = global_subsystem_instance(PythonRepos)
                           python_setup = global_subsystem_instance(PythonSetup)
                           interpreter_cache = PythonInterpreterCache(python_setup, python_repos)
                           interpreter = interpreter_cache.select_interpreter_for_targets(targets)
                         
                           pythonpath = [os.path.join(get_buildroot(), t.target_base) for t in targets]
                           for dist in resolve(['thrift=={}'.format(self.get_thrift_version(apache_thrift_gen))],
                                               interpreter=interpreter,
                                               context=python_repos.get_network_context(),
                                               fetchers=python_repos.get_fetchers()):
                             pythonpath.append(dist.location)
                         
                           process = subprocess.Popen([interpreter.binary,
                                                       '-c',
                                                       'from foo.bar.ttypes import One; from foo.baz.ttypes import Two'],
                                                      env={'PYTHONPATH': os.pathsep.join(pythonpath)},
                                                      stderr=subprocess.PIPE)
                           _, stderr = process.communicate()
                     >     self.assertEqual(0, process.returncode, stderr)
                     E     AssertionError: Traceback (most recent call last):
                     E       File "<string>", line 1, in <module>
                     E       File "/private/var/folders/1j/jzw90nc51nn_358mmhnwc_fh0000gn/T/tmp4mubUC_BUILD_ROOT/.pants.d/pants_backend_codegen_thrift_python_apache_thrift_py_gen_ApacheThriftPyGen/252d64521cf9/src.thrift.com.foo.one/current/foo/__init__.py", line 1, in <module>
                     E         __import__('pkg_resources').declare_namespace(__name__)
                     E     ImportError: No module named pkg_resources
                     
                     .pants.d/pyprep/sources/954a06f59059fc6632440c9815cc316fff4bff3a/pants_test/backend/codegen/thrift/python/test_apache_thrift_py_gen.py:165: AssertionError
                     =============== 2 tests deselected ===============
                     ==== 1 failed, 2 deselected in 13.53 seconds =====
                     
                   tests/python/pants_test/backend/codegen/thrift/python                           .....   FAILURE
FAILURE: FAILURE


               Waiting for background workers to finish.
13:28:52 01:00   [complete]
               FAILURE
```

--- a/tests/python/pants_test/backend/codegen/thrift/python/test_apache_thrift_py_gen.py
+++ b/tests/python/pants_test/backend/codegen/thrift/python/test_apache_thrift_py_gen.py
@@ -149,7 +149,13 @@ class ApacheThriftPyGenTest(TaskTestBase):
     interpreter_cache = PythonInterpreterCache(python_setup, python_repos)
     interpreter = interpreter_cache.select_interpreter_for_targets(targets)
 
-    pythonpath = [os.path.join(get_buildroot(), t.target_base) for t in targets]
+    # We need setuptools to import namespace packages (via pkg_resources), so we prime the
+    # PYTHONPATH with interpreter extras, which Pants always populates with setuptools and wheel.
+    # TODO(John Sirois): We really should be emitting setuptools in a
+    # `synthetic_target_extra_dependencies` override in `ApacheThriftPyGen`:
+    #   https://github.com/pantsbuild/pants/issues/5975
+    pythonpath = interpreter.extras.values()
+    pythonpath.extend(os.path.join(get_buildroot(), t.target_base) for t in targets)
     for dist in resolve(['thrift=={}'.format(self.get_thrift_version(apache_thrift_gen))],
                         interpreter=interpreter,
                         context=python_repos.get_network_context(),

